File: server/Services/CultMembersService.cs



namespace instaCult.Services;


public class CultMembersService(CultMembersRepository repo, CultsService cultsService)
{
    private readonly CultMembersRepository repo = repo;
    private readonly CultsService cultsService = cultsService;
    // NOTE a cultMember DOES depend on cults existing, so we can inject CULT type dependencies here

    internal CultMemberProfile CreateCultMember(CultMember cultMemberData)
    {
        Cult cult = cultsService.GetCultById(cultMemberData.CultId);
        // REVIEW idk is there any other rules we want to implement here? maybe i have to be the owner or something?
        CultMemberProfile cultist = repo.CreateCultMember(cultMemberData);
        return cultist;
    }

    internal List<CultMemberProfile> GetCultMembersByCultId(int cultId)
    {
      Cult cult = cultsService.GetCultById(cultId);
      // REVIEW idk is there any other rules we want to implement here? maybe i have to be the owner or something?
      List<CultMemberProfile> cultists = repo.GetCultMembersByCultId(cultId);
      return cultists;
      
    }

    internal List<CultViewModel> GetMyCults(string userId)
    {
      List<CultViewModel> myCults = repo.GetMyCults(userId);
      return myCults;
    }
}

File: client/src/services/CultMembersService.js

import {api} from './AxiosService.js'
import {logger} from '../utils/Logger.js'
import { CultMemberProfile } from '../models/Account.js'
import { AppState } from '../AppState.js'
import { CultViewModel } from '../models/Cult.js'

class CultMembersService{
async getCultMembers(cultId){
  const response = await api.get(`api/cults/${cultId}/cultMembers`)
  logger.log('🧌🧌', response.data)
  const cultists = response.data.map(cultMemberProfile => new CultMemberProfile(cultMemberProfile))
  AppState.activeCultMembers = cultists
}

async joinCult(cultMemberData){
  const response = await api.post('api/cultMembers', cultMemberData)
  logger.log('🧌✨', response.data)
  const cultist = new CultMemberProfile(response.data)
  AppState.activeCultMembers.push(cultist)
}

async getMyCults(){
  const response = await api.get('account/cultMembers')
  logger.log('👥🔒', response.data)
  const myCults = response.data.map(cultVM => new CultViewModel(cultVM))
  AppState.accountCults = myCults
}
}

export const cultMembersService = new CultMembersService()
